tests/cpu/fileformats/xmlutils/XMLReaderUtils_tests.cpp

```diff
@@ -43,15 +43,12 @@ OCIO_ADD_TEST(XMLReaderHelper, string_to_float_failure)
     // 2 characters are parsed and this is the length required.
     OCIO_CHECK_NO_THROW(OCIO::ParseNumber(str1, 0, 2, value));
 
-    const char str2[] = "12345";
-    const size_t len2 = std::strlen(str2);
-    // All characters are parsed and this is more than the required length.
-    // The string to double function strtod does not stop at a given length,
-    // but we detect that strtod did read too many characters.
-    OCIO_CHECK_THROW_WHAT(OCIO::ParseNumber(str2, 0, len2 - 2, value),
-                          OCIO::Exception,
-                          "followed by unexpected characters");
```

Comment:
On least one macOS & AppleClang 10, the code above is now working with the last proposal. Did you face a problem?

Reply:
If enabled, this happens: https://github.com/amyspark/OpenColorIO/actions/runs/1453341456

I need to cherry-pick the last commit to fix two typos in the ci-workflow changes.

Reply:
Based on the link, the code throws an exception as expected but the unit test code does not expect one! 
The right behavior is throw an exception, and the original unit test code handles it. 

The change is to uncomment lines 46 to 53 (in red above), and remove lines 47 to 51 (in green below). Is it what you did?

Reply:
Here's the commit: https://github.com/amyspark/OpenColorIO/commit/95e199a2f5253f5a307fc98365d3ed0d1fdec576

The real issue is that, test-wise, there is a difference between the two implementations. That's why tests without fast-float throw an exception (correctly) and tests with fast-float do not (correctly as well).

Reply:
The changes in the commit are not the ones to do. By reverting to the original code it should work.

The original code is:
```
    const char str2[] = "12345";
    const size_t len2 = std::strlen(str2);
    // All characters are parsed and this is more than the required length.
    // The string to double function strtod does not stop at a given length,
    // but we detect that strtod did read too many characters.
    OCIO_CHECK_THROW_WHAT(OCIO::ParseNumber(str2, 0, len2 - 2, value),
                          OCIO::Exception,
                          "followed by unexpected characters");
```

Reply:
> The changes in the commit are not the ones to do. By reverting to the original code it should work.

It will work, but you'll instead break the tests when fast_float is enabled. Since those will correctly parse "123" with the specified boundaries, no exception will be thrown.

Reply:
Yep. No good solution as `fast_float::from_chars` & `strtof` behavior differently. 
Concerning the test in discussion, the problem is the `strtof` implementation only. So, you can protect these few lines with an `#ifdef`.